src/components/Questions/index.js

```diff
@@ -9,45 +9,37 @@ import Answers from '@components/Answers';
 import Modal from '@components/Modal';
 import QuestionForm from '@components/QuestionForm';
 import Button from '@components/Button';
+import Searchbar from '../Searchbar';
 
 function Questions() {
   const { id } = useParams();
   const [questions, setQuestions] = useState([]);
   const [filteredQuestions, setFilteredQuestions] = useState([]);
   const [searchInput, setSearchInput] = useState('');
   const [isShowingModal, setIsShowingModal] = useState(false);
-  const [isShowingMoreQuestions, setIsShowingMoreAnswers] = useState(false);
+  const [count, setCount] = useState(4);
 
   useEffect(async () => {
-    const { data } = await getQuestionsByProductId(id);
-    console.log('🚀 ~ file: index.js ~ line 26 ~ useEffect ~ data', data);
-    setQuestions(data.results);
-    setFilteredQuestions(data.results);
-  }, []);
+    async function fetchData() {
+      const { data } = await getQuestionsByProductId(id, count);
+      setQuestions(data.results);
+      setFilteredQuestions(data.results);
+    }
+    fetchData();
+  }, [id, count]);
 
   useEffect(() => {
     if (searchInput.length >= 3) {
       const filteredQuestions = questions.filter((q) => {
```

Comment:
since you arent going to use `filteredQuestions` outside of passing it into `setFilteredQuestions` i would opt to just include the entirety of your logic within the invocation of `setFilteredQuestions`

```node.js
setFilteredQuestions(
  questions.filter(
    ({ question_body: qBody }) => qBody.match(new RegExp(searchInput, 'gi'))
  )
)
```

😎